x509/src/certificate.rs

```diff
@@ -21,9 +24,65 @@ pub fn default_zero() -> u32 {
     0
 }
 
-/// only support v3 certificates
+/// only supporting v3 certificates, but all versions are listed here
 /// Version  ::=  INTEGER  {  v1(0), v2(1), v3(2)  }
-pub const X509_CERT_VERSION: u8 = 2;
+
+/// Version identifier for X.509 certificates. In practice, only v3 is used.
+#[derive(Copy, Clone, Debug, PartialEq, Eq, PartialOrd)]
+pub enum Version {
+    /// Denotes X.509 v1
+    V1 = 0,
+
+    /// Denotes X.509 v2 - added issuerUniqueID and subjectUniqueID (both of which have been deprecated)
+    V2 = 1,
+
+    /// Denotes X.509 v3 - add extensions
+    V3 = 2,
+}
+
+impl Default for Version {
+    fn default() -> Self {
+        Version::V3
+    }
+}
+
+impl DecodeValue<'_> for Version {
+    fn decode_value(decoder: &mut Decoder<'_>, header: Header) -> der::Result<Self> {
+        Version::try_from(u8::decode_value(decoder, header)?).map_err(|_| Self::TAG.value_error())
+    }
+}
+
+impl EncodeValue for Version {
+    fn value_len(&self) -> der::Result<der::Length> {
+        Ok(Length::ONE)
+    }
+
+    fn encode_value(&self, encoder: &mut Encoder<'_>) -> der::Result<()> {
+        u8::from(*self).encode_value(encoder)
+    }
+}
+
+impl FixedTag for Version {
+    const TAG: Tag = Tag::Integer;
+}
+
+impl From<Version> for u8 {
+    fn from(version: Version) -> Self {
+        version as u8
+    }
+}
+
+impl TryFrom<u8> for Version {
+    type Error = Error;
+    fn try_from(byte: u8) -> Result<Version, Error> {
+        match byte {
+            0 => Ok(Version::V1),
+            1 => Ok(Version::V2),
+            2 => Ok(Version::V3),
+            _ => Err(Self::TAG.value_error()),
+        }
+    }
+}
```

Comment:
@carl-wallace All the changes in this file are obsolete if we merge #410 instead.

Reply:
Sure. Further review of this until after alignment with changes to dependencies is likely not worthwhile.